src/highlight/HighlightAnnotations.tsx

```diff
@@ -1,45 +1,170 @@
 import * as React from 'react';
-import HighlightCanvas from './HighlightCanvas';
+import noop from 'lodash/noop';
 import HighlightCreator from './HighlightCreator';
 import HighlightList from './HighlightList';
-import { AnnotationHighlight } from '../@types';
-import { isValidHighlight, sortHighlight } from './highlightUtil';
-
+import PopupReply from '../components/Popups/PopupReply';
+import SingleHighlightAnnotation from './SingleHighlightAnnotation';
+import { AnnotationHighlight, Rect } from '../@types';
+import { CreatorHighlight, CreatorStatus, Mode } from '../store';
 import './HighlightAnnotations.scss';
 
+type CreateArg = {
+    location: number;
+    message: string;
+    shapes: Rect[];
+    text?: string;
+};
+
 type Props = {
     activeAnnotationId: string | null;
     annotations: AnnotationHighlight[];
+    createHighlight?: (arg: CreateArg) => void;
     isCreating: boolean;
+    location: number;
+    message: string;
     setActiveAnnotationId: (annotationId: string | null) => void;
+    setMessage: (message: string) => void;
+    setMode: (mode: Mode) => void;
+    setStaged: (staged: CreatorHighlight | null) => void;
+    setStatus: (status: CreatorStatus) => void;
+    staged?: CreatorHighlight | null;
+    status: CreatorStatus;
 };
 
-export default class HighlightAnnotations extends React.PureComponent<Props> {
+type State = {
+    highlightRef?: HTMLAnchorElement;
+};
+
+export function getStagedAnnotation(
+    staged?: CreatorHighlight | null,
+): Pick<AnnotationHighlight, 'id' | 'target'> | null {
+    if (!staged) {
+        return null;
+    }
+
+    const { location, shapes } = staged;
+
+    return {
+        id: 'staged',
```

Comment:
Why staged needs an id? Can we simply `SingleHighlightAnnotation` to just render rects like `RegionRect` in `RegionAnnotations`?

Reply:
Probably don't need the id. Highlight annotations are slightly different because it's the canvas that renders the visual treatment and the svg anchor that has a position that can be used as a reference for the popupreply